buildconfig/CMake/Eigen.cmake

```diff
@@ -8,7 +8,8 @@ option(USE_SYSTEM_EIGEN "Use the system installed Eigen - v${eigen_version}?" OF
 
 if(USE_SYSTEM_EIGEN)
   message(STATUS "Using system Eigen")
-  find_package(Eigen ${eigen_version} EXACT REQUIRED)
+  find_package(Eigen3 ${eigen_version} EXACT REQUIRED
```

Comment:
I was discussing this with @peterfpeterson, and were thinking to leave it as it is as the download is small and it will ensure the correct version, as opposed to the developer making sure it is compatible. I think either approach should be fine, and in either case I expect it would be rarely used.